internal/tests/api/with-ns-registered/minipool_test.go

```diff
@@ -420,6 +422,79 @@ func TestMinipoolDeposit(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, types.MinipoolStatus_Staking, mp.Common().Status.Formatted())
 	t.Log("Minipool is in staking")
+
+	// Fast forward time
+	slotsToAdvance = 1200 * 60 * 60 / 12
+	err = testMgr.AdvanceSlots(uint(slotsToAdvance), false)
+	require.NoError(t, err)
+	t.Logf("Advanced %d slots", slotsToAdvance)
+
+	var wethBalanceYieldDistributorBefore *big.Int
+	err = qMgr.Query(func(mc *batch.MultiCaller) error {
+		weth.BalanceOf(mc, &wethBalanceYieldDistributorBefore, csMgr.YieldDistributor.Address)
+		return nil
+	}, nil)
+	require.NoError(t, err)
+
+	// Send ETH to YieldDistributor to trigger finalizeInterval
+	// sendResponse, err := hd.Wallet.Send(eth.EthToWei(1), "eth", csMgr.YieldDistributor.Address)
+	// require.NoError(t, err)
+	// MineTx(t, sendResponse.Data.TxInfo, sendEthOpts, "Sent ETH to the YieldDistributor")
+
+	sendEthTx := eth.TransactionInfo{To: csMgr.YieldDistributor.Address, Value: eth.EthToWei(1)}
+	sendEthTx.SimulationResult.IsSimulated = true
+	sendEthOpts, _ := bind.NewKeyedTransactorWithChainID(deployerKey, big.NewInt(int64(chainID)))
+	sendEthOpts.Value = eth.EthToWei(1)
+	MineTx(t, &sendEthTx, sendEthOpts, "Sent ETH to the YieldDistributor")
+
+	// Mine the tx
+	err = testMgr.CommitBlock()
+	require.NoError(t, err)
+
+	var wethBalanceYieldDistributorAfter *big.Int
+	err = qMgr.Query(func(mc *batch.MultiCaller) error {
+		weth.BalanceOf(mc, &wethBalanceYieldDistributorAfter, csMgr.YieldDistributor.Address)
+		return nil
+	}, nil)
+	require.NoError(t, err)
+
+	require.Equal(t, 1, wethBalanceYieldDistributorAfter.Cmp(wethBalanceYieldDistributorBefore))
+
+	// Get the node address
+	nodeKey, err := keygen.GetEthPrivateKey(4)
+	require.NoError(t, err)
+	nodePubkey := crypto.PubkeyToAddress(nodeKey.PublicKey)
+
+	// Get wrapped ETH balance of node pub key before harvest
+	var wethBalanceNodeBefore *big.Int
+	var wethBalanceTreasuryBefore *big.Int
+
+	err = qMgr.Query(func(mc *batch.MultiCaller) error {
+		weth.BalanceOf(mc, &wethBalanceNodeBefore, nodePubkey)
+		weth.BalanceOf(mc, &wethBalanceTreasuryBefore, treasuryAddress)
+		return nil
+	}, nil)
+	require.NoError(t, err)
+
+	// Make a harvest TX for the minipool
+	harvestTxInfo, err := csMgr.YieldDistributor.Harvest(nodePubkey, common.Big0, common.Big1, deployerOpts)
+	require.NoError(t, err)
+	require.NotNil(t, harvestTxInfo)
+	MineTx(t, harvestTxInfo, deployerOpts, "Harvested minipool")
+
+	// Get wrapped ETH balance of node pub key after harvest
+	var wethBalanceNodeAfter *big.Int
+	var wethBalanceTreasuryAfter *big.Int
+
+	err = qMgr.Query(func(mc *batch.MultiCaller) error {
+		weth.BalanceOf(mc, &wethBalanceNodeAfter, nodePubkey)
+		weth.BalanceOf(mc, &wethBalanceTreasuryAfter, treasuryAddress)
+		return nil
+	}, nil)
+	require.NoError(t, err)
+
+	require.Equal(t, 1, wethBalanceNodeAfter.Cmp(wethBalanceNodeBefore))
+	require.Equal(t, 1, wethBalanceTreasuryAfter.Cmp(wethBalanceTreasuryBefore))
```

Comment:
Probably a good idea to put a log line at the end here for confirmation that they were, indeed, bigger. For stuff like this I tend to just print out the before and after balances directly so I can see them as part of the test.

Reply:
added